src/liberty/devCommands.ts

```diff
@@ -145,6 +145,40 @@ export async function startContainerDevMode(libProject?: LibertyProject | undefi
     }
 }
 
+export async function buildStarterProject( state?: any, libProject?: LibertyProject | undefined): Promise<void> {
+    var apiURL = `https://start.openliberty.io/api/start?a=${state.a}&b=${state.b}&e=${state.e}&g=${state.g}&j=${state.j}&m=${state.m}`;
+    const downloadStarterProject = async function(downloadLocation: string): Promise<void> {
+        axios({
+        method: "get",
+        url: apiURL,
+        responseType: "stream"
+        }).then( function (response){
+            response.data.pipe(fs.createWriteStream(downloadLocation))
+            .on("close", () => {
+                var unzip = require("unzip-stream");
+                fs.createReadStream(downloadLocation).pipe(unzip.Extract({ path: `${state.dir}/${state.a}`}));
+                fs.unlink(downloadLocation, async (err) => {
+                    const folderUri = vscode.Uri.file(state.dir);
+                    if (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders[0].uri.fsPath != state.dir) {
```

Comment:
Is the idea here that if the workspace is empty we automatically add the newly generated project to the current workspace?

Reply:
Yes, if it's empty or the target dir is the same as the currrent, then we don't ask to open new window.